frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -161,30 +160,32 @@ public void setTestResponse(EntityModel<Response> testResponse) {
     public UploadImageModel(final Guid limitToStorageDomainId, final DiskImage resumeUploadDisk) {
         if (resumeUploadDisk == null) {
             setDiskModel(new NewDiskModel() {
+                private final boolean isChangeable = limitToStorageDomainId == null;
+
                 @Override
                 public void initialize() {
                     super.initialize();
 
-                    getStorageDomain().setIsChangeable(limitToStorageDomainId == null);
-                    getDataCenter().setIsChangeable(limitToStorageDomainId == null);
+                    getDataCenter().setIsChangeable(isChangeable);
```

Comment:
You wrote at the PR description: `By initializing the unchangeable data center and storage domain once the model loads for the first time ...`
Doesn't it mean that you need to have this line `getDataCenter().setIsChangeable(isChangeable);` **after** line 172 (after `if (!isChangeable)`)?

Reply:
DC is being initialized in super.initialize() > commonInitialize() > updateDatacenters() > getDataCenter().setSelectedItem(...)

The SD update is being triggered only on updateStorageDomains(), as part of datacenter_SelectedItemChanged(), but the field is still without a selected value (as mentioned in prev. comment)